src/chainparams.h

```diff
@@ -101,7 +103,7 @@ class CChainParams
     std::vector<SeedSpec6> vFixedSeeds;
     bool fDefaultConsistencyChecks;
     bool fRequireStandard;
-    bool fMineBlocksOnDemand;
+    bool m_user_can_change_chain_params;
```

Comment:
Policies are not really chain params...

Reply:
> you can't change chain params, they're a unique_ptr<const ..> so the compiler would complain if you even tried...

That sounds like an implementation detail to me. Effectively you can change the chain param: The const `fRequireStandard` in the chain params global serves as default and the global `fRequireStandard` is the effective value. I think having two globals with the same name (in different namespaces) is confusing and should go, but I am not doing this here.

My opinion is that user should be able to change chain params whenever they are running a test chain (e.g. testnet, regtest or testchains #8994 by @jtimon). And they should be forbidden to change them when they are running on mainnet.

Reply:
The name of the variable is also an implementation detail :)

The way I look at it is ChainParams for mainnet/testnet/regtest is effectively a compile-time constant that includes consensus parameters, various policy defaults, and some constraints on user's ability to change policy (this one, allow_fallback_fee). You can't change any of those things (without recompiling). That you can choose something different to a default doesn't mean you're changing the default though, which is what you'd be doing if you were actually changing the chain params. Just like you're not changing `DEFAULT_BLOCK_MAX_WEIGHT` when you set `-blockmaxweight`, eg.

Even in a testchains world, I think many of the chain params for testnet (or public signets) probably should remain hard to change -- changing the address prefixes/hrp or default port or network magic or consensus parameters when you don't know what you're doing seems like a great way to shoot yourself in the foot, for instance. (If you do know what you're doing, obviously you can change the code to do what you want and recompile)

Reply:
Ok makes sense, but I still fail to see why the `m_user_can_change_chain_params` member is wrong. I could call it `m_user_can_override_chain_params` (as opposed to overwrite/change) to make clear that chainparams are the default values, that can potentially be overriden?

Reply:
If it let you override any chain param, that name would make sense; but it only lets you override this specific one. I don't think it makes much sense to try to make it more general; if it did, combining `m_fallback_fee_enabled` into it ought to work, but I think just having separate bools enabling the individual features is clearer.

Reply:
My point was only that this isn't a chain param at all. The default value for the policy just happens to reside on the chainparams object as an implementation detail.

Reply:
My goal is to remove the wallet-specific `m_fallback_fee_enabled` from the chain params later on. I can rename `m_user_can_change_chain_params` to `m_is_test_chain`, which should address both concerns raised in the comments here.

Reply:
Note that #8994 maintains the ```unique_ptr<const ..>``` and the chainparams can never be changed after initialization, not even for custom chains.

Reply:
Ok, removed the "user can change chain params" field.